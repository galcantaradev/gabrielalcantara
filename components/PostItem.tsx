import Link from 'next/link';
import styled from 'styled-components';

import { SectionTitle } from './SectionTitle';
import { FormattedDate } from './FormattedDate';
import { IPost } from '../types';
import { formattedTags } from '../utils';

interface Props {
  post: IPost;
}

const Article = styled.article`
  margin-bottom: 3rem;

  p {
    margin: 0;
  }

  h1 {
    color: ${props => props.theme.green};
    margin-bottom: 0.5rem;
    width: fit-content;

    :hover {
      cursor: pointer;
    }
  }
`;

export const PostItem = (props: Props) => {
  const { post } = props;

  return (
    <Article>
      <Link href={`/posts/${post.id}`}>
        <SectionTitle>{post.title}</SectionTitle>
      </Link>
      <small>
        ☕️ {post.readTime} {post.readTime > 1 ? 'mins' : 'min'} de leitura •{' '}
        <FormattedDate value={post.createdAt} />
      </small>
      <p>{post.description}</p>
      <small>🏷 {formattedTags(post.tags)}</small>
    </Article>
  );
};
