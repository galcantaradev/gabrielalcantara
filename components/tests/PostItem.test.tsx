import { render, screen } from '@testing-library/react';

import { PostItem } from '../PostItem';
import { IPost } from '../../types';

const post: IPost = {
  id: '1',
  title: 'Post Title',
  description: 'Post Description',
  readTime: 0,
  markdown: 'Post Markdown',
  tags: ['react', 'jest', 'react testing library'],
  createdAt: '2021-01-01'
};

describe('PostItem.test.tsx', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<PostItem post={post} />);

    expect(screen.getByText('Post Title')).toBeInTheDocument();
    expect(screen.getByText('Post Description')).toBeInTheDocument();
    expect(
      screen.getByText('☕️ 0 min de leitura • 01/01/2021')
    ).toBeInTheDocument();
    expect(
      screen.getByText('🏷 react, jest, react testing library')
    ).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it('read time should be singular', () => {
    render(<PostItem post={post} />);

    expect(
      screen.getByText('☕️ 0 min de leitura • 01/01/2021')
    ).toBeInTheDocument();
  });

  it('read time should be plural', () => {
    render(<PostItem post={{ ...post, readTime: 2 }} />);

    expect(
      screen.getByText('☕️ 2 mins de leitura • 01/01/2021')
    ).toBeInTheDocument();
  });
});
