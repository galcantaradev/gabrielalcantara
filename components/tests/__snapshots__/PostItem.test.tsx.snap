// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostItem.test.tsx should render correctly 1`] = `
<DocumentFragment>
  <article
    class="PostItem__Article-v55yx6-0 gnRnNm"
  >
    <h1
      class="SectionTitle-sc-15x45w6-0 fzzHJf"
    >
      Post Title
    </h1>
    <small>
      ☕️ 0 min de leitura • 01/01/2021
    </small>
    <p>
      Post Description
    </p>
    <small>
      🏷 react, jest, react testing library
    </small>
  </article>
</DocumentFragment>
`;
